### PR TITLE
Catalog Book class

### PR DESCRIPTION
**Pull request by raminka13**
### Task #1 #7 #8 #9 #10 
- Created Book class in a separate .rb file.
- Created Label class with an association to the Item class (in a separate .rb file).
- All Book class properties visible in the diagram should be defined and set up in the constructor method.
- All Label class properties visible in the diagram should be defined and set up in the constructor method.
- Implemented methods:
    - add_item method in the Label class
        - Takes an instance of the Item class as an input
        - Adds the input item to the collection of items
        - Adds self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Book class
        - Overrides the method from the parent class
        - Returns true if parent's method returns true OR if cover_state equals to "bad"
        - otherwise, it should return false
- Added unit tests for all implemented methods.
- The following options should be available:
    - List all books
    - List all labels (e.g. 'Gift', 'New')
    - Add a book
- All data should be preserved by saving collections in .json files.

### CODE REVIEW

File: book/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  context '#move_to_archive' do
+    it "should return true if cover_state equals 'bad', even if parent's method returns false" do
+      bad_book = Book.new('publisher', 'bad')
+      expect(bad_book.move_to_archive).to be true
+    end
+
+    # it "should return true if the parent's method returns true" do
+    #   expect(@book.move_to_archive).to be true
+    # end
+
```
Comment (Nyame-Wolf) [inline book/book_spec.rb:51]: Kindly remove commented code. It makes your code look unprofessional.

**Review by Nyame-Wolf — APPROVED** [review]
# STATUS: APPROVED :heavy_check_mark: :green_circle: 

Hi @raminka13  :smile: , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :sparkles:

## Optional suggestions :ballot_box_with_check: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

APPROVED! 💯 💯 💯

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me, @Nyame-Wolf in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book/book_spec.rb (hunk 1 above)
Comment (Fourteen98) [inline book/book_spec.rb:51]: if you don't need it, kindly remove it. you can always go back in commit.

**Review by Fourteen98 — APPROVED** [review]
Hi @raminka13  😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Descriptive pull request 
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**